src/shared/user-record-show.c

```diff
@@ -131,8 +131,26 @@ void user_record_show(UserRecord *hr, bool show_full_group_info) {
                         printf(" Password OK: %sno%s (%m)\n", ansi_highlight_yellow(), ansi_normal());
                         break;
                 }
-
-                printf(" Password OK: %syes%s\n", ansi_highlight_green(), ansi_normal());
+                if (strv_isempty(hr->hashed_password)) {
+                        printf(" Password OK: %sno%s (none set)\n", ansi_highlight(), ansi_normal());
+                        break;
+                }
+                if (strv_contains(hr->hashed_password, "")) {
+                        printf(" Password OK: %sno%s (none set)\n",
```

Comment:
this should say "empty set".

(i.e. the block above covers the case where there are zero passwords set, i.e. the account is not accessible via password at all – maybe because SSH auth should be used exclusively. This block should cover the case where an *empty* password is set, i.e. you get logged in without providing any. Thus we should make the difference clear in the output text and color)

Reply:
Thanks I got it

Reply:
If we have to make the difference, what color should I output this in ? @poettering

Reply:
zero passwords set is not an issue, a non-accessible account is a good account hence maybe just highlight it (`ansi_highlight()`), since it's interesting but nothing beyond that. Having an empty password set is an issue likely, since it means anyone can access the account just like that, hence mark it red (`ansi_highlight_red()`), or so?